web/src/hotkey.js

```diff
@@ -733,15 +733,29 @@ export function process_hotkey(e, hotkey) {
         return false;
     }
 
-    if (hotkey.message_view_only && popover_menus.is_gear_menu_popover_displayed()) {
-        // Inside the gear menu, we don't process most hotkeys; the
-        // exception is that the gear_menu hotkey should toggle the
-        // menu closed again.
+    // We don't need to process arrow keys in navbar menus for spectators
+    // since they only have gear menu present.
+    if (
+        popover_menus.is_personal_menu_popover_displayed() &&
+        event_name === "left_arrow" &&
+        !page_params.is_spectator
+    ) {
+        // Open gear menu popover on left arrow.
+        $("#personal-menu").trigger("click");
+        gear_menu.toggle();
+        return true;
+    }
+
+    if (popover_menus.is_gear_menu_popover_displayed()) {
         if (event_name === "gear_menu") {
             gear_menu.toggle();
             return true;
+        } else if (event_name === "right_arrow" && !page_params.is_spectator) {
+            // Open personal menu popover on g + right arrow.
+            gear_menu.toggle();
+            $("#personal-menu").trigger("click");
```

Comment:
This seems like it's intended to just toggle both menus; feels a bit weird that the toggles are expressed so differently given they are both tippy popovers. I guess also -- do the first two checks in `toggle` actually do something now that it's a Tippy popover?

```
export function toggle() {                                                                              
    if (popover_menus.is_gear_menu_popover_displayed()) {                                               
        popovers.hide_all();                                                                            
        return;                                                                                         
    }                                                                                                   
                                                                                                        
    // Since this can be called via hotkey, we need to                                                  
    // hide any other popovers that may be open before.                                                 
    if (popovers.any_active()) {                                                                        
        popovers.hide_all();                                                                            
    }                                                                                                   
                                                                                                        
    $("#gear-menu").trigger("click");                                                                   
}                                                                                                       
```

Reply:
> do the first two checks in toggle actually do something now that it's a Tippy popover?

Yes, since tippy doesn't handle hiding other popovers via hotkeys and we are doing a `trigger("click")` which tippy doesn't use to hide existing popovers.

Alternative is to call `trigger("mousedown")`  that tippy does handle to hide all popover and `trigger("click") but it is just simpler to call hide_all.

Reply:
OK.